.github/scripts/build-macos.sh

```diff
@@ -1,12 +1,20 @@
 #!/bin/bash
 set -e
 
+if [[ "${GITHUB_REF_TYPE}" == "tag" ]]; then
+    BUILD_TAG=${GITHUB_REF_NAME}
+else 
+    BUILD_TAG=$(git rev-parse --short HEAD)
+fi
+
 # install deps
 brew install SDL2
 
 # build
-cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_TESTS=ON -DDETHRACE_WERROR=ON -B build
+cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_OSX_ARCHITECTURES=x86_64 -DBUILD_TESTS=ON -DDETHRACE_WERROR=ON -B build
 cmake --build build -- -j 4
 
 # package artifact
-tar -czvf darwin-amd64.tar.gz build/dethrace
+tar -czvf dethrace-${BUILD_TAG}-darwin-amd64.tar.gz build/dethrace
+
+echo "::set-output name=filename::dethrace-${BUILD_TAG}-darwin-amd64.tar.gz"
```

Comment:
Something what bothers me with the current Linux release is that the tarball contains a `build` folder.
How about we create a `dethrace-${BUILD_TAG}-darwin-amd64` directory, and tar that directory?

```suggestion
releasename="dethrace-${BUILD_TAG}-darwin-amd64"

rm -rf "$releasename"
mkdir "$releasename"
cp build/dethrace "$releasename/dethrace"
tar -czvf "$releasename.tar.gz" "$releasename"

echo "::set-output name=filename::"$releasename.tar.gz"
```

Reply:
The same applies to the Macos release binary.
The Windows zip contains no directories.